python/appledoublekaitai/kaitai_utils.py

```diff
@@ -1,62 +1,200 @@
 import enum
+import importlib
 import inspect
 import json
+import os
 from io import BufferedWriter
 from typing import Any, BinaryIO, Dict, Generator, List, Type
 
-import kaitaistruct
-from kaitaistruct import KaitaiStruct
+import hansken_extraction_plugin
+import yaml
+from hansken_extraction_plugin.api.extraction_trace import ExtractionTrace
+from hansken_extraction_plugin.api.transformation import Range, RangedTransformation
 from json_stream import streamable_dict, streamable_list
+from kaitaistruct import KaitaiStruct
 
 
-def write_to_json(data_binary: BinaryIO, writer: BufferedWriter, class_type: Type[KaitaiStruct]):
+def write_kaitai_to_trace(trace: ExtractionTrace, max_byte_array_length: int):
     """
-    Writes a binary form of JSON string into a BufferedWriter
-
-    @param data_binary: binary data containing the file content
-    @param writer: bufferedWriter to write the binary form of the JSON string to
-    @param class_type: class that contains the parsing to a KaiTai struct
-    @return: JSON string representing contents of data object
+    This method creates traces from a kaitai object tree. It results in a trace containing a JSON representation of the
+    object tree and potential child traces containing byte arrays.
+    @param trace: ExtractionTrace provided to the plugin. Child traces are added to this.
```

Comment:
```suggestion

    @param trace: ExtractionTrace provided to the plugin. Child traces are added to this.
```

Reply:
This file was copied from the template, it should be changed there... Will create a separate PR for that.